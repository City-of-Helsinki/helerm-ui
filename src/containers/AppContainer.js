import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Provider, connect } from 'react-redux';
import { Router } from 'react-router-dom';
import ReduxToastr from 'react-redux-toastr';

import Loader from '../components/Loader';
import { retrieveUserFromSession } from '../components/Login/reducer';
import { fetchAttributeTypes, fetchTemplates } from '../store/uiReducer';

class AppContainer extends Component {
  static propTypes = {
    fetchAttributeTypes: PropTypes.func.isRequired,
    fetchTemplates: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    retrieveUserFromSession: PropTypes.func.isRequired,
    routes: PropTypes.object.isRequired,
    store: PropTypes.object.isRequired,
    user: PropTypes.object
  };

  UNSAFE_componentWillMount() {
    this.props.fetchAttributeTypes();
    this.props.fetchTemplates();
    this.props.retrieveUserFromSession();
  }

  render() {
    const { user, routes, store, history } = this.props;
    return (
      <Provider store={store}>
        <div style={{ height: '100%' }}>
          {user ? (
            <Router history={history}>{routes}</Router>
          ) : (
            <Loader show={true} />
          )}
          <ReduxToastr
            timeOut={4000}
            newestOnTop={true}
            preventDuplicates={true}
            position='top-right'
            transitionIn='fadeIn'
            transitionOut='bounceOutUp'
            progressBar={true}
          />
        </div>
      </Provider>
    );
  }
}

const mapStateToProps = ({ user }) => ({
  user: user.data
});

const mapDispatchToProps = {
  fetchAttributeTypes,
  fetchTemplates,
  retrieveUserFromSession
};

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
