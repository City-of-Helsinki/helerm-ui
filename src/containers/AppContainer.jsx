/* eslint-disable camelcase */
/* eslint-disable react/forbid-prop-types */
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Provider, connect } from 'react-redux';
import { Router } from 'react-router-dom';
import ReduxToastr from 'react-redux-toastr';
import { LoginProvider, SessionEndedHandler, useOidcClient } from 'hds-react';

import { retrieveUserFromSession } from '../components/Login/reducer';
import { fetchAttributeTypes, fetchTemplates } from '../store/uiReducer';
import { providerProperties } from '../utils/oidc/constants';

const App = ({
  history,
  routes,
  dispatchRetrieveUserFromSession,
  dispatchFetchAttributeTypes,
  dispatchFetchTemplates,
}) => {
  const { getUser, isAuthenticated } = useOidcClient();

  const isAuth = isAuthenticated();
  const user = getUser();

  useEffect(() => {
    dispatchFetchAttributeTypes();
    dispatchFetchTemplates();

    if (isAuth) {
      const { profile } = user;
      const { sub: userId } = profile;

      dispatchRetrieveUserFromSession(userId);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isAuth, user]);

  return (
    <div style={{ height: '100%' }}>
      <SessionEndedHandler
        content={{
          title: 'Istunto on vanhentunut!',
          text: 'Istuntosi palvelimella on vanhentunut. Sinut kirjataan ulos palvelusta.',
          buttonText: 'Kirjaudu ulos',
          closeButtonLabelText: 'Kirjaudu ulos',
        }}
      />
      <Router history={history}>{routes}</Router>
      <ReduxToastr
        timeOut={4000}
        newestOnTop
        preventDuplicates
        position='top-right'
        transitionIn='fadeIn'
        transitionOut='bounceOutUp'
        progressBar
      />
    </div>
  );
};

const AppContainer = ({
  routes,
  store,
  history,
  dispatchRetrieveUserFromSession,
  dispatchFetchAttributeTypes,
  dispatchFetchTemplates,
}) => (
  <LoginProvider {...providerProperties}>
    <Provider store={store}>
      <App
        history={history}
        routes={routes}
        dispatchRetrieveUserFromSession={dispatchRetrieveUserFromSession}
        dispatchFetchAttributeTypes={dispatchFetchAttributeTypes}
        dispatchFetchTemplates={dispatchFetchTemplates}
      />
    </Provider>
  </LoginProvider>
);

App.propTypes = {
  history: PropTypes.object.isRequired,
  routes: PropTypes.node.isRequired,
  dispatchFetchAttributeTypes: PropTypes.func.isRequired,
  dispatchFetchTemplates: PropTypes.func.isRequired,
  dispatchRetrieveUserFromSession: PropTypes.func.isRequired,
};

AppContainer.propTypes = {
  dispatchFetchAttributeTypes: PropTypes.func.isRequired,
  dispatchFetchTemplates: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  dispatchRetrieveUserFromSession: PropTypes.func.isRequired,
  routes: PropTypes.node.isRequired,
  store: PropTypes.object.isRequired,
};

const mapDispatchToProps = {
  dispatchRetrieveUserFromSession: retrieveUserFromSession,
  dispatchFetchAttributeTypes: fetchAttributeTypes,
  dispatchFetchTemplates: fetchTemplates,
};

export default connect(null, mapDispatchToProps)(AppContainer);
