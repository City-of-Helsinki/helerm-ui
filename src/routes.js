import React from 'react';
import { Switch, Route } from 'react-router-dom';
import CoreLayout from './layouts/CoreLayout/CoreLayout';
import InfoLayout from './layouts/InfoLayout/InfoLayout';
import BulkListViewContainer from './components/Bulk/BulkListViewContainer';
import BulkCreateViewContainer from './components/Bulk/BulkCreateView/BulkCreateViewContainer';
import BulkViewContainer from './components/Bulk/BulkView/BulkViewContainer';
import FacetedSearchContainer from './components/FacetedSearch/FacetedSearchContainer';
import ViewTOSContainer from './components/Tos/ViewTos/ViewTosContainer';
import ViewClassificationContainer from './components/Classification/ViewClassification/ViewClassificationContainer';
import ClassificationTreeContainer from './components/ClassificationTree/ClassificationTreeContainer';
import ViewInfo from './components/Info/ViewInfo';
import PrintTOS from './components/Tos/Print/PrintView';
import NotFound from './components/NotFound/NotFound';
import LoginCallbackContainer from './components/LoginCallback/LoginCallbackContainer'
import RenewCallbackContainer from './components/RenewCallback/RenewCallbackContainer'

const Routes = () => {
  return (
    <Switch>
      <Route exact path='/info'>
        <InfoLayout>
          <ViewInfo />
        </InfoLayout>
      </Route>
      <Route exact path='/callback'>
        <InfoLayout>
          <LoginCallbackContainer />
        </InfoLayout>
      </Route>
      <Route exact path='/renew'>
        <RenewCallbackContainer />
      </Route>
      <Route exact path='/bulk'>
        <InfoLayout />
        <BulkListViewContainer />
      </Route>
      <Route exact path='/bulk/view/:id'>
        <InfoLayout />
        <BulkViewContainer />
      </Route>
      <Route exact path='/bulk/create'>
        <InfoLayout />
        <BulkCreateViewContainer />
      </Route>
      <Route exact path={['/', '/filter']}>
        <CoreLayout>
          <ViewInfo />
        </CoreLayout>
      </Route>
      <Route exact path='/search'>
        <CoreLayout />
        <FacetedSearchContainer />
      </Route>
      <Route exact path={['/view-tos/:id/', '/view-tos/:id/version/:version']}>
        <CoreLayout />
        <ViewTOSContainer />
      </Route>
      <Route
        exact
        path={['/view-tos/:id/print', '/view-tos/:id/version/:version/print']}
      >
        <CoreLayout />
        <PrintTOS />
      </Route>
      <Route
        exact
        path={[
          '/view-classification/:id',
          '/view-classification/:id/version/:version'
        ]}
      >
        <CoreLayout />
        <ViewClassificationContainer />
      </Route>
      <Route exact path='/classification-tree'>
        <CoreLayout />
        <ClassificationTreeContainer />
      </Route>
      <Route path='*' component={NotFound} />
    </Switch>
  );
};

export default Routes;
