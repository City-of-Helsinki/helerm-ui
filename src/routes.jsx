import React from 'react';
import { Route, Navigate } from 'react-router-dom';

import CoreLayout from './layouts/CoreLayout/CoreLayout';
import InfoLayout from './layouts/InfoLayout/InfoLayout';
import LoginCallbackContainer from './views/LoginCallback/LoginCallbackContainer';
import NotFound from './components/NotFound/NotFound';
import ErrorPage from './views/ErrorPage/ErrorPage';

const ViewInfo = React.lazy(() => import('./views/ViewInfo/ViewInfo'));
const BulkListViewContainer = React.lazy(() => import('./views/Bulk/BulkListViewContainer'));
const BulkViewContainer = React.lazy(() => import('./views/Bulk/BulkView/BulkViewContainer'));
const BulkCreateViewContainer = React.lazy(() => import('./views/Bulk/BulkCreateView/BulkCreateViewContainer'));
const FacetedSearchContainer = React.lazy(() => import('./views/FacetedSearch/FacetedSearchContainer'));
const ViewTOSContainer = React.lazy(() => import('./views/Tos/ViewTos/ViewTosContainer'));
const PrintTOS = React.lazy(() => import('./views/Tos/Print/PrintView'));
const ViewClassificationContainer = React.lazy(() => import('./views/ViewClassification/ViewClassificationContainer'));
const ClassificationTreeContainer = React.lazy(() => import('./views/ClassificationTree/ClassificationTreeContainer'));
const CookieManagement = React.lazy(() => import('./views/CookieManagement/CookieManagement'));

const RoutesComponent = () => (
  <>
    <Route
      exact
      path='/info'
      element={
        <InfoLayout>
          <ViewInfo />
        </InfoLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/callback'
      element={
        <InfoLayout>
          <LoginCallbackContainer />
        </InfoLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/bulk'
      element={
        <InfoLayout>
          <BulkListViewContainer />
        </InfoLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/bulk/view/:id'
      element={
        <InfoLayout>
          <BulkViewContainer />
        </InfoLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/bulk/create'
      element={
        <InfoLayout>
          <BulkCreateViewContainer />
        </InfoLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/'
      element={
        <CoreLayout>
          <ViewInfo />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/filter'
      element={
        <CoreLayout>
          <ViewInfo />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/search'
      element={
        <CoreLayout>
          <FacetedSearchContainer />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/view-tos/:id/'
      element={
        <CoreLayout>
          <ViewTOSContainer />
        </CoreLayout>
      }
    />
    <Route
      exact
      path='/view-tos/:id/version/:version'
      element={
        <CoreLayout>
          <ViewTOSContainer />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/view-tos/:id/print'
      element={
        <CoreLayout>
          <PrintTOS />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/view-tos/:id/version/:version/print'
      element={
        <CoreLayout>
          <PrintTOS />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/view-classification/:id'
      element={
        <CoreLayout>
          <ViewClassificationContainer />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/view-classification/:id/version/:version'
      element={
        <CoreLayout>
          <ViewClassificationContainer />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/classification-tree'
      element={
        <CoreLayout>
          <ClassificationTreeContainer />
        </CoreLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route
      exact
      path='/cookies'
      element={
        <InfoLayout>
          <CookieManagement />
        </InfoLayout>
      }
      errorElement={<Navigate to='/error' />}
    />
    <Route exact path='/error' element={<ErrorPage />} />
    <Route path='*' element={<NotFound />} />
  </>
);

export default RoutesComponent;
