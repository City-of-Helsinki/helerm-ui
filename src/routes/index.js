import React from 'react';
import { Route, IndexRoute } from 'react-router';
import CoreLayout from '../layouts/CoreLayout/CoreLayout';
import Home from './UI/containers/HomeContainer';
import ViewTOSContainer from './ViewTOS/containers/ViewTOSContainer';

export default () => (
  <Route path='/' component={CoreLayout}>
    <IndexRoute component={Home} />
    <Route path='view-tos/:id' component={ViewTOSContainer} />
  </Route>
);
