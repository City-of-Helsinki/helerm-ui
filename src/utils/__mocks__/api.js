function get() {
  return Promise.resolve({});
}

function post() {
  return Promise.resolve({});
}

function put() {
  return Promise.resolve({});
}

function patch() {
  return Promise.resolve({});
}

function del() {
  return Promise.resolve({});
}

const methods = { get, post, put, patch, del };

export default methods;
