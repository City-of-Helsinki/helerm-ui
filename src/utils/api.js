import fetch from 'isomorphic-fetch';
import Promise from 'promise-polyfill';
import {
  forEach,
  merge
} from 'lodash';

import { getStorageItem, removeStorageItem } from './storage';

/**
 * Which actions are allowed without authentication
 * @type {[*]}
 */
const ALLOWED_METHODS_WITHOUT_AUTHENTICATION = ['GET'];

// Add Promise to window if not supported...
if (!window.Promise) {
  window.Promise = Promise;
}

/**
 *
 * @param endpoint
 * @param params
 * @param options
 * @returns {*}
 */
export function get (endpoint, params = {}, options = {}) {
  return callApi(endpoint, params, options);
}

/**
 *
 * @param endpoint
 * @param data
 * @param params
 * @param options
 * @returns {*}
 */
export function post (endpoint, data, params = {}, options = {}) {
  if (typeof data !== 'string') {
    data = JSON.stringify(data);
    options.headers = merge(
      { 'Content-Type': 'application/json' },
      options.headers
    );
  }
  return callApi(endpoint, params, merge({ body: data, method: 'POST' }, options));
}

/**
 *
 * @param endpoint
 * @param data
 * @param params
 * @param options
 * @returns {*}
 */
export function put (endpoint, data, params = {}, options = {}) {
  if (typeof data !== 'string') {
    data = JSON.stringify(data);
    options.headers = merge(
      { 'Content-Type': 'application/json' },
      options.headers
    );
  }
  return callApi(endpoint, params, merge({ body: data, method: 'PUT' }, options));
}

export function patch (endpoint, data, params = {}, options = {}) {
  if (typeof data !== 'string') {
    data = JSON.stringify(data);
    options.headers = merge(
      { 'Content-Type': 'application/json' },
      options.headers
    );
  }
  return callApi(endpoint, params, merge({ body: data, method: 'PATCH' }, options));
}

/**
 *
 * @param endpoint
 * @param params
 * @param options
 * @returns {*}
 */
export function del (endpoint, params = {}, options = { method: 'DELETE' }) {
  return callApi(endpoint, params, options);
}

/**
 *
 * @param endpoint
 * @param params
 * @param options
 * @returns {*}
 */
export function callApi (endpoint, params, options = {}) {
  const token = getStorageItem('token');
  const defaultHeaders = new Headers();
  const url = getApiUrl(endpoint, params);
  const finalOptions = merge({
    method: 'GET',
    credentials: 'include',
    mode: 'cors'
  }, options);

  defaultHeaders.append('Accept', 'application/json');

  if (!token && ALLOWED_METHODS_WITHOUT_AUTHENTICATION.indexOf(finalOptions.method) !== 0) {
    throw Error(`Following methods for API-endpoint require authentication: ${ALLOWED_METHODS_WITHOUT_AUTHENTICATION.join(', ')}`);
  }

  if (token) {
    defaultHeaders.append('Authorization', `JWT ${token}`);
  }

  if (options.headers) {
    Object.keys(options.headers).map(key => {
      defaultHeaders.append(key, options.headers[key]);
    });
  }

  finalOptions.headers = defaultHeaders;
  return fetch(url, finalOptions)
    .then(res => {
      // TODO: Remove me & use refresh token
      if (res.status === 401) {
        removeStorageItem('token', () => {
          return window.location.reload();
        });
      }
      return res;
    });
}

/**
 *
 * @param url
 * @param query
 * @returns {string}
 */
export function getApiUrl (url, query = {}) {
  const queryString = buildQueryString(query);
  return [API_URL, API_VERSION, url, queryString].join('/');
}

/**
 *
 * @param query
 * @returns {string}
 */
export function buildQueryString (query) {
  const pairs = [];

  forEach(query, (value, key) => {
    pairs.push([key, value].join('='));
  });

  return pairs.length ? '?' + pairs.join('&') : '';
}

export default { get, post, put, patch, del };
