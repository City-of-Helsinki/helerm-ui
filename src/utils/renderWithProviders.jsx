/* eslint-disable react/prop-types */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { LoginProvider } from 'hds-react';

import { initialState as navigationInitialState } from '../store/reducers/navigation';
import { initialState as validationInitialState } from '../store/reducers/validation';
import { initialState as routerInitialState } from '../store/reducers/router';
import { initialState as selectedTOSInitialState } from '../store/reducers/tos-toolkit/main';
import { initialState as classificationInitialState } from '../store/reducers/classification';
import { initialState as userInitialState } from '../store/reducers/user';
import { initialState as uiInitialState } from '../store/reducers/ui';
import { initialState as bulkInitialState } from '../store/reducers/bulk';
import { initialState as searchInitialState } from '../store/reducers/search';
import storeCreator from '../store/createStore';

export const storeDefaultState = {
  navigation: navigationInitialState,
  validation: validationInitialState,
  router: routerInitialState,
  selectedTOS: selectedTOSInitialState,
  classification: classificationInitialState,
  user: userInitialState,
  ui: uiInitialState,
  bulk: bulkInitialState,
  search: searchInitialState,
};

const renderWithProviders = (
  ui,
  { preloadedState = storeDefaultState, store = storeCreator(preloadedState), ...renderOptions } = {},
) => {
  const Wrapper = ({ children }) => (
    <LoginProvider>
      <Provider store={store}>{children}</Provider>
    </LoginProvider>
  );

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export default renderWithProviders;
