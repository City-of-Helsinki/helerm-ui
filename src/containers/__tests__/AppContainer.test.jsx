import React from 'react';
import { createBrowserHistory } from 'history';
import * as mockLogin from 'hds-react';

import renderWithProviders from '../../utils/renderWithProviders';
import AppContainer from '../AppContainer';
import routes from '../../routes';
import storeCreator from '../../store/createStore';
import mockUser from '../../utils/mocks/user.json';
import api from '../../utils/api';

const getUserMock = vitest.fn().mockImplementation(() => ({ profile: { sub: 'user123' } }));
const isAuthenticatedMock = vitest.fn().mockImplementation(() => true);
const mockToken = 'mockToken';
const mockApiGet = vitest.fn().mockImplementation(() => Promise.resolve({ ok: true, json: () => mockUser }));

vitest.spyOn(mockLogin, 'useOidcClient').mockImplementation(() => ({
  getUser: getUserMock,
  isAuthenticated: isAuthenticatedMock,
}));
vitest.spyOn(mockLogin, 'getApiTokenFromStorage').mockImplementation(() => mockToken);
vitest.spyOn(api, 'get').mockImplementation(mockApiGet);

const renderComponent = (history) => {
  const mockState = {};
  const mockStore = storeCreator(history, mockState);
  const mockRoutes = routes(mockStore);

  return renderWithProviders(
    <AppContainer
      history={history}
      routes={mockRoutes}
      store={mockStore}
      dispatchFetchAttributeTypes={vitest.fn()}
      dispatchFetchTemplates={vitest.fn()}
      dispatchRetrieveUserFromSession={vitest.fn()}
    />,
    {
      history,
      store: mockStore,
    },
  );
};

describe('<AppContainer />', () => {
  it('should render correctly', () => {
    const history = createBrowserHistory();

    const { container } = renderComponent(history);

    expect(container).toMatchSnapshot();
  });

  it('should authenticate', () => {
    const history = createBrowserHistory();

    renderComponent(history);

    expect(getUserMock).toHaveBeenCalled();
    expect(isAuthenticatedMock).toHaveBeenCalled();
    expect(mockApiGet).toHaveBeenCalled();
  });

  it('should not fetch user if not authenticated', () => {
    const storageSpy = vitest.spyOn(mockLogin, 'getApiTokenFromStorage').mockImplementationOnce(() => undefined);

    vitest.spyOn(mockLogin, 'useOidcClient').mockImplementationOnce(() => ({
      getUser: vitest.fn().mockImplementation(() => undefined),
      isAuthenticated: vitest.fn().mockImplementation(() => false),
    }));

    const history = createBrowserHistory();

    renderComponent(history);

    expect(storageSpy).not.toHaveBeenCalled();
  });
});
