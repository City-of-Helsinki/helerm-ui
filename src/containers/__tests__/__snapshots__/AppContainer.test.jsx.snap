// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppContainer /> > should render correctly 1`] = `
<div>
  <div
    style="height: 100%;"
  >
    <div
      class="core-layout__viewport"
    >
      <div
        class="header"
      >
        <nav
          class="navbar navbar-inverse container-fluid"
          style="background-color: rgb(0, 114, 198);"
        >
          <a
            class="brand-title navbar-brand logo"
            href="/"
          >
            <svg
              viewBox="0 0 495 230"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M479.8 14.9v131.5c0 25.1-20.7 45.6-46.2 45.6H292.9c-16.9 0-32.9 6.1-45.4 17.4C235 198.2 219 192 202.1 192h-141c-25.3 0-45.8-20.4-45.8-45.6V14.9H479.8zM255 221.9c10.1-10.1 23.6-15.7 38-15.7h140.6c33.3 0 60.4-26.8 60.4-59.8V0.6H1v145.7c0 33 26.9 59.8 60.1 59.8h141c14.3 0 27.8 5.6 38 15.7l7.5 7.5L255 221.9z"
              />
              <path
                d="M425.7 74.8c0 4.4-3.5 7.6-7.9 7.6s-7.9-3.2-7.9-7.6 3.5-7.6 7.9-7.6C422.2 67 425.7 70.2 425.7 74.8M410.9 136.9h13.7V88.2h-13.7V136.9zM386.3 107.5l17.9 29.4h-14.9l-12-19.5 -6.5 8.2v11.3h-13.6v-70h13.6v32.3c0 6-0.7 12-0.7 12h0.3c0 0 3.2-5.3 5.9-9.1l10.4-13.9h16L386.3 107.5zM343.4 105.7c0-12.1-5.9-18.7-15.3-18.7 -8 0-12.2 4.7-14.6 9.2h-0.3l0.7-8h-13.6v48.7h13.6V108c0-5.6 3.3-9.5 8.4-9.5s7.3 3.4 7.3 9.9v28.5h13.7L343.4 105.7 343.4 105.7zM287.2 74.8c0 4.4-3.5 7.6-7.9 7.6s-7.9-3.2-7.9-7.6 3.5-7.6 7.9-7.6C283.8 67 287.2 70.2 287.2 74.8M272.4 136.9h13.7V88.2h-13.7V136.9zM248.8 123.3c0-3.9-5.4-4.7-11.7-6.5 -7.3-1.9-16.5-5.6-16.5-14.9 0-9.2 8.8-15 19.2-15 9.7 0 18.6 4.6 22.3 11.1l-11.7 6.6c-1.2-4.3-4.8-7.7-10.2-7.7 -3.5 0-6.4 1.5-6.4 4.4 0 3.7 6.4 4.1 13.6 6.4 7.6 2.4 14.6 5.9 14.6 14.8 0 9.8-9.1 15.6-19.6 15.6 -11.4 0-20.4-4.8-24.4-12.6l11.9-6.7c1.5 5.3 5.8 9.3 12.3 9.3C246.1 128.1 248.8 126.5 248.8 123.3M205.5 67.3h-13.7v56.1c0 5 1.1 8.7 3.2 11 2.2 2.3 5.6 3.4 10.3 3.4 1.6 0 3.3-0.2 4.9-0.5 1.7-0.4 3-0.9 4.1-1.5l1.2-9.2c-1.1 0.4-2 0.7-2.9 0.8 -0.8 0.2-1.8 0.2-2.9 0.2 -1.8 0-2.6-0.5-3.2-1.4 -0.7-0.9-1-2.5-1-4.8V67.3zM159.9 96.8c-5 0-9.1 3.6-10.2 10.2h19.2C169 101.2 165.2 96.8 159.9 96.8M181 116.2h-31.5c0.6 7.9 5 11.8 10.7 11.8 5.1 0 8.6-3.4 9.4-8l11.5 6.5c-3.4 6.2-10.7 11.6-20.9 11.6 -13.6 0-23.6-9.5-23.6-25.5 0-15.9 10.2-25.5 23.3-25.5s21.8 9.3 21.8 22.5C181.6 113.5 181 116.2 181 116.2M111.3 136.9h14.2V69.9h-14.2v26.5H84.6V70H70.4v66.9h14.2v-27.7h26.7V136.9z"
              />
            </svg>
          </a>
          <a
            class="brand-title navbar-brand"
            href="/"
          >
            Tiedonohjaus
          </a>
          <p
            class="navbar-text"
          >
            <small>
              
               
              123
            </small>
          </p>
          <p
            class="navbar-text pull-right login-link"
          >
            <button
              class="btn btn-link login-button"
              type="button"
            >
              Kirjaudu ulos
            </button>
          </p>
          <p
            class="navbar-text pull-right"
          >
            <a
              class="navbar-link"
              href="/search"
            >
              Haku
            </a>
          </p>
          <p
            class="navbar-text pull-right"
          >
            <a
              class="navbar-link"
              href="https://hel.fi"
              rel="noreferrer"
              target="_blank"
            >
              Anna palautetta
            </a>
          </p>
        </nav>
        <div
          class="loader-container"
        >
          <span
            class="fa-solid fa-2x fa-spinner fa-spin loader"
          />
        </div>
      </div>
      <div
        class="core-layout__navigation"
      >
        <div
          class="helerm-navigation"
        >
          <div
            class="navigation-menu navigation-open"
            id="navigation-menu"
          >
            <div
              style="min-height: 0px;"
            >
              <div
                class=" "
                style="transform: translateZ(0);"
              >
                <div
                  class="navigation-header clearfix"
                >
                  <button
                    class="pull-left nav-button"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      class="fa-solid fa-list"
                    />
                  </button>
                </div>
                <div
                  class="navigation-container"
                >
                  <div
                    class="navigation-toggle"
                  >
                    <button
                      class="nav-button"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                        class="fa-solid fa-xmark"
                      />
                    </button>
                  </div>
                  <div
                    class="navigation-filters clearfix"
                  >
                    <div
                      class="navigation-filters-container"
                    >
                      <div
                        class="row"
                      >
                        <div
                          class="col-sm-6"
                        >
                          <input
                            class="react-infinity-menu-default-search-input"
                            placeholder="Etsi..."
                            type="search"
                            value=""
                          />
                        </div>
                        <div
                          class="col-sm-6"
                        >
                          <div
                            class=""
                          >
                            <div
                              class=""
                            >
                              <div
                                class="Select css-2b097c-container"
                              >
                                <div
                                  class=" css-yk16xz-control"
                                >
                                  <div
                                    class=" css-g1d714-ValueContainer"
                                  >
                                    <div
                                      class=" css-1wa3eu0-placeholder"
                                    >
                                      Suodata tilan mukaan...
                                    </div>
                                    <div
                                      class="css-b8ldur-Input"
                                    >
                                      <div
                                        class=""
                                        style="display: inline-block;"
                                      >
                                        <input
                                          aria-autocomplete="list"
                                          autocapitalize="none"
                                          autocomplete="off"
                                          autocorrect="off"
                                          id="react-select-2-input"
                                          spellcheck="false"
                                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                                          tabindex="0"
                                          type="text"
                                          value=""
                                        />
                                        <div
                                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: inherit; letter-spacing: normal; text-transform: none;"
                                        />
                                      </div>
                                    </div>
                                  </div>
                                  <div
                                    class=" css-1hb7zxy-IndicatorsContainer"
                                  >
                                    <span
                                      class=" css-1okebmr-indicatorSeparator"
                                    />
                                    <div
                                      aria-hidden="true"
                                      class=" css-tlfecz-indicatorContainer"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="css-6q0nyr-Svg"
                                        focusable="false"
                                        height="20"
                                        viewBox="0 0 20 20"
                                        width="20"
                                      >
                                        <path
                                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                        />
                                      </svg>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="classification-link"
                    >
                      <a
                        class="btn btn-primary btn-sm"
                        href="/classification-tree"
                      >
                        <span
                          aria-hidden="true"
                          class="fa-solid fa-info"
                        />
                      </a>
                    </div>
                  </div>
                  <div
                    class="infinity-menu-container"
                  >
                    <div
                      class="infinity-menu-display-tree-container"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container-fluid helerm-content"
        style="display: none;"
      />
      <div
        class="loader-container"
      >
        <span
          class="fa-solid fa-2x fa-spinner fa-spin loader"
        />
      </div>
    </div>
    <div
      aria-live="assertive"
      class="redux-toastr"
    >
      <div>
        <div
          class="top-left"
        />
        <div
          class="top-right"
        />
        <div
          class="top-center"
        />
        <div
          class="bottom-left"
        />
        <div
          class="bottom-right"
        />
        <div
          class="bottom-center"
        />
      </div>
    </div>
  </div>
</div>
`;
